lib/sanford/connection_handler.rb

```diff
@@ -38,47 +34,52 @@ def run
     protected
 
     def run!
-      service = ProcessedService.new
+      processed_service = ProcessedService.new
       begin
         request = Sanford::Protocol::Request.parse(@connection.read_data)
         self.log_request(request)
-        service.request = request
+        processed_service.request = request
 
         route = @server_data.route_for(request.name)
         self.log_handler_class(route.handler_class)
-        service.handler_class = route.handler_class
+        processed_service.handler_class = route.handler_class
 
         response = route.run(request, @server_data)
-        service.response = response
+        processed_service.response = response
       rescue StandardError => exception
-        self.handle_exception(service, exception, @server_data)
+        self.handle_exception(exception, @server_data, processed_service)
       ensure
-        self.write_response(service)
+        self.write_response(processed_service)
       end
-      service
+      processed_service
     end
 
-    def write_response(service)
+    def write_response(processed_service)
       begin
-        @connection.write_data service.response.to_hash
+        @connection.write_data processed_service.response.to_hash
       rescue StandardError => exception
-        service = self.handle_exception(service, exception)
-        @connection.write_data service.response.to_hash
+        processed_service = self.handle_exception(
+          exception,
+          @server_data,
+          processed_service
+        )
+        @connection.write_data processed_service.response.to_hash
       end
       @connection.close_write
-      service
+      processed_service
     end
 
-    def handle_exception(service, exception, server_data = nil)
-      error_handler = Sanford::ErrorHandler.new(
-        exception,
-        server_data,
-        service.request
-      )
-      service.response  = error_handler.run
-      service.exception = error_handler.exception
-      self.log_exception(service.exception)
-      service
+    def handle_exception(exception, server_data, processed_service)
+      error_handler = Sanford::ErrorHandler.new(exception, {
+        :server_data   => server_data,
+        :request       => processed_service.request,
+        :handler_class => processed_service.handler_class,
+        :response      => processed_service.response
```

Comment:
@jcredding need to support the `:response` key in the error context handling.  looks like it is ignored right now.